UnitTests/UnitTests.Shared/Mvvm/Test_Messenger.cs

```diff
@@ -511,6 +511,29 @@ void Test()
             messenger.Cleanup();
         }
 
+        [TestCategory("Mvvm")]
+        [TestMethod]
+        [DataRow(typeof(StrongReferenceMessenger))]
+        [DataRow(typeof(WeakReferenceMessenger))]
+        public void Test_Messenger_RegisterMultiple_UnregisterSingle(Type type)
+        {
+            var messenger = (IMessenger)Activator.CreateInstance(type);
+
+            var recipient1 = new object();
+            var recipient2 = new object();
+
+            int handlerCalledCount = 0;
+
+            messenger.Register<object, MessageA>(recipient1, (r, m) => { handlerCalledCount++; });
+            messenger.Register<object, MessageA>(recipient2, (r, m) => { handlerCalledCount++; });
+
+            messenger.UnregisterAll(recipient2);
+
+            messenger.Send(new MessageA());
+
+            Assert.AreEqual(1, handlerCalledCount);
```

Comment:
What about tracking each handler separately to double check not only the total number of handlers being raised, but also that the right one is actually being invoked? The current test is already fine, but just in case 😄

```suggestion
            int handler1CalledCount = 0;
            int handler2CalledCount = 0;

            messenger.Register<object, MessageA>(recipient1, (r, m) => { handler1CalledCount++; });
            messenger.Register<object, MessageA>(recipient2, (r, m) => { handler2CalledCount++; });

            messenger.UnregisterAll(recipient2);

            messenger.Send(new MessageA());

            Assert.AreEqual(1, handler1CalledCount);
            Assert.AreEqual(0, handler2CalledCount);
```